### DISCUSSION TITLE
Python Row Factory?

### DISCUSSION THREAD

**Discussion by tERyceNzAchE** [duckdb/duckdb#5441] [Ideas]
Is there any plan to for a row factory in the Python implementation similar to that of SQLite row_factory?
It would allow results to be accessed by dict key/value instead of index.

**Comment by davetapley** [comment]
I'd be happy to make a PR, but I don't know where to start as it looks like all the `.py` [are generated](https://github.com/duckdb/duckdb/blob/8ff5885ee65bcf294948b5a85a9ec16477cb9758/tools/pythonpkg/duckdb-stubs/__init__.pyi) (see also https://github.com/duckdb/duckdb/discussions/5729) 🤔 

Here's a workaround I'm using:

```py
def fetchone_dict(res: DuckDBPyResult):
    keys = [d[0] for d in res.description()]
    values = res.fetchone()
    return dict(zip(keys, values))
```

See also: [SQLite reference](https://docs.python.org/3/library/sqlite3.html#sqlite3-howto-row-factory).
